app/controllers/users_controller.rb

```diff
@@ -21,15 +21,22 @@ def contact_info
   end
 
   def get_preferences
-    response = echo_client.get_preferences(get_user_id, token)
-    render json: response.body, status: response.status
+    preferences_response = echo_client.get_preferences(get_user_id, token, echo_client, session[:access_token])
+
+    urs_response = echo_client.get_urs_user(session[:user_name], session[:access_token])
+    if urs_response.status == 200
+      preferences_response.body['preferences']['general_contact'] = urs_response.body
+    else
+      preferences_response.body['preferences']['general_contact'] = {}
+    end
+
+    render json: preferences_response.body, status: response.status
   end
 
-  def update_contact_info
+  def update_notification_pref
     preferences = {preferences: params.delete('preferences')}
-    user_id = get_user_id
-
-    response = echo_client.update_preferences(user_id, preferences, token)
+    preferences[:preferences]['general_contact']['role'] = "Order Contact"
+    response = echo_client.update_preferences(get_user_id, preferences, token)
```

Comment:
If I'm reading this correctly, you're relying on getting the whole preference object from the client but only updating the notification preference. Can you use get_preferences to avoid having to get all that information?